api/src/main/java/org/eclipse/microprofile/config/spi/ConfigNode.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2017, 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.eclipse.microprofile.config.spi;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Marker interface identifying a config node implementation.
+ */
+public interface ConfigNode {
+    /**
+     * Key of this config node.
+     *
+     * @return key of this node
+     */
+    String key();
+
+    /**
+     * Get the type of this node.
+     *
+     * @return NodeType this node represents
+     */
+    NodeType nodeType();
```

Comment:
Some time, in the far, far future, the spec will require a minimum of Java 17. Should we prepare for this day by declaring the type enum to be a temporary workaround until this interface can be `sealed` in a future version?